tests/foreman/cli/test_activationkey.py

```diff
@@ -1149,6 +1150,122 @@ def test_positive_remove_host_collection_by_name(self):
+        self.assertEqual(new_ak['system-purpose']['purpose-role'], "test-role")
+        self.assertEqual(new_ak['system-purpose']['purpose-usage'], "test-usage")
+        if not is_open('BZ:1789028'):
+            self.assertEqual(new_ak['system-purpose']['service-level'], "Self-Support")
+        # Check that system purpose values can be deleted.
+        ActivationKey.update({
+                    u'id': new_ak['id'],
+                    u'purpose-addons': '',
+                    u'purpose-role': '',
+                    u'purpose-usage': '',
+                    u'service-level': '',
+                    u'organization-id': self.org['id'],
+                })
+        updated_ak = ActivationKey.info({'id': new_ak['id'], 'organization-id': self.org['id']})
+        self.assertEqual(updated_ak['system-purpose']['purpose-addons'], '')
+        self.assertEqual(updated_ak['system-purpose']['purpose-role'], '')
+        self.assertEqual(updated_ak['system-purpose']['purpose-usage'], '')
+        self.assertEqual(updated_ak['system-purpose']['service-level'], '')
+
+    @tier2
+    def test_update_ak_with_syspurpose_values(self):
+        """Test that system purpose values can be added to an existing activation key
+        and can then be changed.
+
+        :id: db943c05-70f1-4385-9537-fe23368a9dfd
+
+        :Steps:
+            1. Create Activation key with no system purpose values set
+            2. Assert system purpose values are not set
+            3. Add system purpose values
+            4. Read the AK system purpose values and assert system purpose values are set
+            5. Update the system purpose values
+            6. Read the AK system purpose values and assert system purpose values have changed
+
+        :CaseImportance: Medium
+
+        :BZ: 1789028
+        """
+        org = make_org()
+        # Create an AK with no system purpose vales set
+        new_ak = self._make_activation_key({
+                    u'organization-id': org['id'],
+                })
+        # We need a manifest to enable service level menu options.
+        # Clone manifest and upload it
+        with manifests.clone() as manifest:
+            upload_file(manifest.content, manifest.filename)
+        try:
+            Subscription.upload({
+                u'file': manifest.filename,
+                u'organization-id': org['id'],
+            })
+        except CLIReturnCodeError as err:
+            raise CLIFactoryError(
+                u'Failed to upload manifest\n{0}'.format(err.msg))
+        # Assert system purpose values are null after creating the AK and adding the manifest.
+        self.assertEqual(new_ak['system-purpose']['purpose-addons'], '')
+        self.assertEqual(new_ak['system-purpose']['purpose-role'], '')
+        self.assertEqual(new_ak['system-purpose']['purpose-usage'], '')
+        self.assertEqual(new_ak['system-purpose']['service-level'], '')
```

Comment:
Do we need that? Is there possibility that uploading manifest will change these values for existing activation key?

Reply:
Adding the manifest makes those values available in the system level drop-down menu.
That does not mean they could be set due to some bug.  I can remove that.